app/src/main/java/com/orange/ods/app/ui/search/SearchScreen.kt

```diff
@@ -16,41 +16,154 @@ import androidx.compose.foundation.lazy.LazyColumn
+import com.orange.ods.app.ui.components.Variant
 import com.orange.ods.app.ui.components.components
+import com.orange.ods.app.ui.guidelines.color.DialogColor
+import com.orange.ods.app.ui.guidelines.color.getValue
+import com.orange.ods.app.ui.guidelines.spacing.Spacing
+import com.orange.ods.app.ui.utilities.DrawableManager
 import com.orange.ods.compose.component.list.OdsListItem
 import com.orange.ods.compose.component.list.OdsListItemIcon
 import com.orange.ods.compose.component.list.OdsListItemIconType
 import com.orange.ods.compose.component.list.iconType
+import com.orange.ods.theme.guideline.GuidelineColor
+import com.orange.ods.theme.guideline.toHexString
 import com.orange.ods.utilities.extension.orElse
 
 @Composable
-fun SearchScreen(searchedText: MutableState<TextFieldValue>, onComponentClick: (Long) -> Unit) {
+fun SearchScreen(searchedText: MutableState<TextFieldValue>, onComponentClick: (Long) -> Unit, onVariantClick: (Long) -> Unit) {
+
+    val context = LocalContext.current
+
     val filterComponents = components.filter { component ->
-        searchedText.value.text.isEmpty() || stringResource(id = component.titleRes).lowercase().contains(searchedText.value.text.lowercase())
+        searchedText.value.text.isEmpty() || stringResource(id = component.titleRes).lowercase()
+            .contains(searchedText.value.text.lowercase())
+    }
+
+    val filterSpacings = Spacing.values().filter { spacing ->
+        searchedText.value.text.isEmpty() || spacing.tokenName.lowercase()
+            .contains(searchedText.value.text.lowercase())
+    }
+
+    val filterGuidelines = LocalOdsGuideline.current.guidelineColors.filter { guidelineColor ->
+        searchedText.value.text.isEmpty() || guidelineColor.getName().lowercase().contains(searchedText.value.text.lowercase()) ||
+                guidelineColor.lightThemeName.lowercase().contains(searchedText.value.text.lowercase()) ||
+                guidelineColor.darkThemeName.lowercase().contains(searchedText.value.text.lowercase())
     }
+
+    val filterVariants = components.filter { it.variants.isNotEmpty() }
+        .flatMap { component ->
+            val componentImageRes = component.smallImageRes.orElse { component.imageRes }
+            component.variants.filter { variant ->
+                searchedText.value.text.isEmpty() || context.getString(variant.titleRes).lowercase()
+                    .contains(searchedText.value.text.lowercase())
+            }.map { variant ->
+                componentImageRes to variant
+            }
+        }
+
+    data class OdsSearchParameter(
+        val title: String,
+        val route: Long,
+        val image: Int?,
+        val subtitle: String?,
+        val color: Color?,
+        val data: Any
+    )
+
+    val searchList: List<OdsSearchParameter> = filterComponents.filter { it.variants.isEmpty() }
+        .map { component ->
+            val componentImageRes = component.smallImageRes.orElse { component.imageRes }
```

Comment:
No nerd to define a new value here cause it is used once. You can directly write:
```kotlin
            OdsSearchParameter(
                context.getString(component.titleRes),
                component.id,
                component.smallImageRes.orElse { component.imageRes },
                color = null,
                subtitle = component.composableName,
                data = component
            )
```